### PR TITLE
Feature/community posts

### PR DESCRIPTION
**Pull request by martinyde**
Most relevant is the itchefer_stream module files

### CODE REVIEW

File: html/modules/custom/itchefer_stream/itchefer_stream.module [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Truncate message post tokens.
+ * @todo Create a new token instead e.g: [message:field_message_related_object:entity:field_post:max150]
+ */
+function itchefer_stream_tokens_alter(array &$replacements, array $context, \Drupal\Core\Render\BubbleableMetadata $bubbleable_metadata) {
+  $token = '[message:field_message_related_object:entity:field_post]';
+  if(isset($context['tokens']['field_post']) && $token === $context['tokens']['field_post']) {
+    if ($replacements[$token] instanceof \Drupal\Core\Render\Markup) {
+      $currentValueStripped = strip_tags($replacements[$token]->__toString());
+      if (strlen($currentValueStripped) > 20) {
+        $replacements[$token] = substr($currentValueStripped, 0, 150) . '...';
```
Comment (rimi-itk) [inline html/modules/custom/itchefer_stream/itchefer_stream.module]: It seems odds to check if the length is more that 20 characters and then stripping to (at most) 150 characters? Shouldn't it be 150 (or 20) in both places?

File: html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+      $related_entity = $this->activityFactory->getActivityRelatedEntity($data);
+      $entity_storage = $this->entityTypeManager->getStorage($related_entity['target_type']);
+      $entity = $entity_storage->load($related_entity['target_id']);
+      $allUsers = $this->entityTypeManager->getStorage('user')->getQuery()->execute();
+      // It could happen that a notification has been queued but the content
+      // has since been deleted. In that case we can find no additional
+      // recipients.
+      if (!$entity) {
+        return $recipients;
+      }
```
Comment (rimi-itk) [inline html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php]: For performance reasons the check for a deleted entity should be performed before getting all users (to get rid of an unnecessary database query for all user).

File: html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+   * {@inheritdoc}
+   */
+  public function isValidEntity(EntityInterface $entity) {
+    // Special cases for comments.
+    if ($entity->getEntityTypeId() === 'comment') {
+      // Returns the entity to which the comment is attached.
+      $entity = $entity->getCommentedEntity();
+    }
+
+    if (!isset($entity)) {
```
Comment (rimi-itk) [inline html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php:114]: This check is inconsistent with the check in line 88. Use `!isset(…)` or `!…` (or `empty(…)` or `NULL === …`) consistently to check to a missing entity.

File: html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      return FALSE;
+    }
+
+    if ($entity->getEntityTypeId() === 'post') {
+      if (!$entity->field_recipient_group->isEmpty()) {
+        return FALSE;
+      }
+      elseif (!$entity->field_recipient_user->isEmpty()) {
+        return FALSE;
+      }
```
Comment (rimi-itk) [inline html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php]: Writing this as 

```php
      if (!$entity->field_recipient_group->isEmpty()
        || !$entity->field_recipient_user->isEmpty()) {
        return FALSE;
      }
```

may make it a little easier to read.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. Some minor issues should be fixed.

Is the change from `langcode: en` to `langcode: da` in config files something that we should worry about?

File: html/modules/custom/itchefer_stream/itchefer_stream.module (hunk 1 above)
Comment (martinyde) [reply html/modules/custom/itchefer_stream/itchefer_stream.module]: Fixed

File: html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php (hunk 2 above)
Comment (martinyde) [reply html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php]: Fixed

File: html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php (hunk 3 above)
Comment (martinyde) [reply html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php:114]: Fixed

File: html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php (hunk 4 above)
Comment (martinyde) [reply html/modules/custom/itchefer_stream/src/Plugin/ActivityContext/CommunityActivityWithRecipientsContext.php]: Fixed

**Comment by martinyde** [comment]
No worries

**Pull request merged**